### ISSUE TITLE
Vite building error (ByteString convertion by undici)

### ISSUE THREAD

**Issue by nemu1113** [kwchang0831/svelte-QWER#34]
After adding a total of 270 markdown files and performing a build, the generated "posts.js" was bloated and the type conversion of characters to bytestring format failed, causing an error, as shown in the log below.

```pwsh
> vite build

vite v3.1.3 building for production...
...(omission)...
transforming (4) node_modules\.pnpm\flexsearch@0.7.21\node_modules\flexsearch\dist\flexsearch.bUse of eval is strongly discouraged, as it poses security risks and may cause issues with minification   
✓ 8 modules transformed.
✓ 885 modules transformed.
[plugin:unocss:global:build:scan] [unocss] failed to find vite:css-post plugin. It might be an internal bug of UnoCSS
rendering chunks (563)...The emitted file "vite-manifest.json" overwrites a previously emitted file 
of the same name.
vite v3.1.3 building SSR bundle for production...
...(omission)...
✓ 3 modules transformed.
No name was provided for external module 'flexsearch' in output.globals – guessing 'flexsearch'
No name was provided for external module 'lz-string' in output.globals – guessing 'LZString'
✓ 441 modules transformed.
[plugin:unocss:global:build:scan] [unocss] failed to find vite:css-post plugin. It might be an internal bug of UnoCSS
.svelte-kit/output/server/vite-manifest.json
                                                  96.70 KiB
.svelte-kit/output/server/index.js
                                                  84.82 KiB
.svelte-kit/output/server/entries/endpoints/atom.xml/_server.ts.js
                                                  2.96 KiB
.svelte-kit/output/server/entries/endpoints/feed.json/_server.ts.js
                                                  1.96 KiB
.svelte-kit/output/server/entries/endpoints/manifest.webmanifest/_server.ts.js
                                                  2.17 KiB
.svelte-kit/output/server/entries/endpoints/sitemap.xml/_server.ts.js
                                                  0.89 KiB
.svelte-kit/output/server/entries/endpoints/versions.json/_server.ts.js
                                                  5.38 KiB
.svelte-kit/output/server/entries/pages/_layout.svelte.js
                                                  22.15 KiB
.svelte-kit/output/server/entries/pages/_layout.ts.js
                                                  0.15 KiB
.svelte-kit/output/server/entries/pages/_error.svelte.js
                                                  1.43 KiB
.svelte-kit/output/server/entries/pages/_page.svelte.js
                                                  13.17 KiB
.svelte-kit/output/server/entries/pages/about/_page.svelte.js
                                                  7.38 KiB
...(omission)...
.svelte-kit/output/server/chunks/index.js
                                                  4.47 KiB
.svelte-kit/output/server/chunks/index2.js
                                                  1.29 KiB
.svelte-kit/output/server/chunks/site.js
                                                  2.87 KiB
.svelte-kit/output/server/chunks/tags.js
                                                  3.81 KiB
.svelte-kit/output/server/chunks/themes.js
                                                  0.78 KiB
.svelte-kit/output/server/chunks/i18n-svelte.js
                                                  1.58 KiB
.svelte-kit/output/server/chunks/image_avatar.svelte_svelte_type_style_lang.js
                                                  6.41 KiB
.svelte-kit/output/server/chunks/image_avatar.js
                                                  0.98 KiB
.svelte-kit/output/server/chunks/assets.js
                                                  3.54 KiB
.svelte-kit/output/server/chunks/post.js
                                                  11.39 KiB
.svelte-kit/output/server/chunks/image_zoom.js
                                                  3.91 KiB
.svelte-kit/output/server/chunks/emoji.js
                                                  0.76 KiB
.svelte-kit/output/server/chunks/gallery.js
                                                  0.80 KiB
.svelte-kit/output/server/chunks/code_copy.js
                                                  0.35 KiB
.svelte-kit/output/server/chunks/info_box.js
                                                  1.68 KiB
.svelte-kit/output/server/chunks/google_map.js
                                                  0.70 KiB
.svelte-kit/output/server/chunks/blog_card.js
                                                  1.35 KiB
.svelte-kit/output/server/chunks/hooks.server.js
                                                  0.24 KiB
.svelte-kit/output/server/chunks/posts.js
                                                  1662.91 KiB
TypeError: Cannot convert argument to a ByteString because the character atindex 20 has a value of 9834 which is greater than 255.
    at Object.webidl.converters.ByteString (C:\Git\qwer\node_modules\.pnpm\undici@5.10.0\node_modules\undici\lib\fetch\webidl.js:403:13)
    at Headers.set (C:\Git\qwer\node_modules\.pnpm\undici@5.10.0\node_modules\undici\lib\fetch\headers.js:362:31)
    at file:///C:/Git/qwer/.svelte-kit/output/server/index.js:2573:29  
    at async Object.handle (file:///C:/Git/qwer/.svelte-kit/output/server/chunks/hooks.server.js:2:46)
    at async respond (file:///C:/Git/qwer/.svelte-kit/output/server/index.js:2562:22)
    at async visit (file:///C:/Git/qwer/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.503_svelte@3.50.1+vite@3.1.3/node_modules/@sveltejs/kit/src/core/prerender/prerender.js:226:20)
file:///C:/Git/qwer/node_modules/.pnpm/@sveltejs+kit@1.0.0-next.503_svelte@3.50.1+vite@3.1.3/node_modules/@sveltejs/kit/src/core/prerender/prerender.js:50
                                throw new Error(format_error(details, config));
                                      ^

```

Is it possible that this is a specification limitation on this theme side?
Alternatively, could it be a problem with sveltekit?
※I don't know where the library "undici" is used.

**Comment by kwchang0831** [comment]
Where is your repo? I would to test it out if I may.

**Comment by nemu1113** [comment]
@kwchang0831
Thank you for your response.
Now I converted my repo to public, so if possible plz check my source.

[My repo](https://github.com/nemu1113/qwer-nemunote)

**Comment by kwchang0831** [comment]
Thanks :D

I will be working on the solution...

**Comment by kwchang0831** [comment]
I am not sure what would be the best soluiton here.
One solution: https://github.com/kwchang0831/qwer-nemunote/tree/68b079b753695cad666bb6004336c296a31341cc

So, the problem is that you are using title as part of the slug / url.
And, as result
1. Post with long title will make the final url too long.
1. Unicode characters will make sveltekit prerendering not happy

Hmm... I need more info to determine what should I do with this problem.

**Comment by nemu1113** [comment]
@kwchang0831 
Thank you for sharing your solution. It was very helpful!!

I see...
I did not have a similar problem with my current Hugo, so I did not specifically implement slug shortening or elimination of unicode characters for each article.
I understood that sveltekit has limitations in that area.

I guess the easiest solution would be to set a slug for each article that consists only of alphanumeric characters and takes string length into account.

**Comment by kwchang0831** [comment]
After more digging, the root cause is located https://github.com/sveltejs/kit/blob/560394f5fded5033e99aebd1774b19cae13172da/packages/kit/src/runtime/server/index.js#L301 when setting up headers and https://github.com/nodejs/undici/blob/main/lib/fetch/webidl.js#L402.

I am not sure what would be the best way to fix.

But what I have tried is 

https://github.com/sveltejs/kit/blob/560394f5fded5033e99aebd1774b19cae13172da/packages/kit/src/runtime/server/index.js#L301
```js 
response.headers.set('x-sveltekit-routeid', encodeURI(event.routeId));
```

https://github.com/sveltejs/kit/blob/560394f5fded5033e99aebd1774b19cae13172da/packages/kit/src/core/prerender/prerender.js#L422
```js
for (const [route_id, prerender] of prerender_map) {
  if (prerender === true && !prerendered_routes.has(encodeURI(route_id))) {
    not_prerendered.push(route_id);
  }
}
```

I need to do more testing and then probably submit a PR to them.

**Comment by kwchang0831** [comment]
So, another solution would be 
https://github.com/kwchang0831/qwer-nemunote/tree/820588e4a7bfe16eb80613a401be251f6bd67c6b

In this solution, I modified sveltekit like I mentioned above.
And, the slug transforms like the following 
`posts/2009/2009-05-05-初めまして.md` -> `/posts/2009/05/04/初めまして` (05/04 due to the published date)

**Comment by kwchang0831** [comment]
But then... if the routeID is already encoded, this fix will double encode it which will cause issue.
Need more digging...

**Comment by kwchang0831** [comment]
@nemu1113 

The unicode routes issue is fixed as of [@sveltejs/kit@1.0.0-next.506](https://github.com/sveltejs/kit/releases/tag/%40sveltejs%2Fkit%401.0.0-next.506)

The [kwchang0831/qwer-nemunote](https://github.com/kwchang0831/qwer-nemunote) repo has been upgraded to latest deps all. 

For the route name, I suggest using filename without the date instead of full title to avoid route being too long. eg: `user/blogs/posts/2009/2009-05-05-初めまして.md` -> `/src/routes/posts/2009/05/04/初めまして`

Since the issue is now resolved, I will go ahead close this issue. If you encouter new problem, please create a new issue. Otherwise, happy coding :D

**Comment by nemu1113** [comment]
@kwchang0831 
Thank you so much!! 
I got it, and I will try on later.
